### PR TITLE
Api calls

### PR DESCRIPTION
**Pull request by surbhi84**
This pr consists of :

- api calls to the backend.
- added color features in result page to show selected answers.
- added feature that if you switch tab or refresh the page the quiz is not continued to prevent dishonesty.

### CODE REVIEW

File: src/apiCalls.ts [hunk 1]
```diff
@@ -33,3 +42,50 @@ export const userSignup = ({
     password,
   });
 };
+
+export const getHighscore = () => {
+  return axios.get<Array<HighscoresModel>>("/highscore");
+};
+
+export const getCategories = () => {
+  return axios.get<Array<CategoryModel>>("/category");
```
Comment (ckahub) [inline src/apiCalls.ts:51]: there is one extra closing bracket here

File: src/userRedux/authSlice.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { useSelector } from "react-redux";
+import { RootState } from "./store";
+
+// const user = useSelector((state: RootState) => state.currentUser.user);
```
Comment (ckahub) [inline src/userRedux/authSlice.ts:5]: remove commented code

**Review by ckahub — APPROVED** [review]
good folder structure
good fromating of code

File: src/backend/controllers/question.controller.ts [hunk 3]
```diff
@@ -28,42 +31,41 @@ export const getQuestionsHandler = function (
         }
       );
     }
-
-    return new Response(200, {}, this.db.questions);
-  } catch (error) {
+    const tags = request.queryParams?.tags?.split(",") ?? [];
```
Comment (rohanmathur91) [inline src/backend/controllers/question.controller.ts:34]: good use of optional chaining

File: src/components/questionCard/QuestionCard.tsx [hunk 4]
```diff
@@ -1,33 +1,170 @@
...
+
+    navigate("/results", {
+      state: {
+        prev: "/quiz",
+        userScore: userScore,
+        questionArray: questions,
+        quesNum: quesNum,
+        selectedArr: [...selectedArr, selected],
+        // category:category,
+      },
```
Comment (rohanmathur91) [inline src/components/questionCard/QuestionCard.tsx:94]: remove commented code

**Review by rohanmathur91 — APPROVED** [review]
- Code is very well written
- Liked the folder structure
- Can't view the preview please check that
- Great work 🔥

**Pull request merged**